multi-agent/lib/mas/sharing/cloner.py

```
@@ -317,10 +324,37 @@ def _clone_single_resource(self, cache_data: ResourceCacheData,
+        """
+        saved: List[Resource] = []
+        try:
+            for doc in docs:
+                self.resources.save_resource(doc)
+                saved.append(doc)
+        except Exception:
+            logger.error(
+                "Batch resource creation failed after saving %d of %d "
+                "resources; rolling back the ones that succeeded",
+                len(saved), len(docs),
+            )
```

Ruff (0.16.0)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)